sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -71,95 +75,45 @@ class DefaultHttpClient implements HttpClient {
      */
     @Override
     public Response<?> send(HttpRequest httpRequest) {
-        if (httpRequest.getHttpMethod() == HttpMethod.PATCH) {
-            return sendPatchViaSocket(httpRequest);
+        HttpConnectionCache.HttpConnection connection = httpConnectionCache.get(new HttpConnectionProperties(httpRequest, httpRequest.getUrl(), proxyOptions));
```

Comment:
The `HttpUrlConnection` implementation in JDK pools the connections (indexed by host and proxy) internally. I was wondering the reason for us to implement pooling/caching in our library.

https://github.com/Azure/azure-storage-java/issues/212#issuecomment-362775663
https://bugs.openjdk.org/browse/JDK-6495923

Reply:
I think we want to look at connection pooling more specifically for socket connections.

Reply:
Right, pooling is useful for Patch+Socket .